tools/syz-db/syz-db.go

```diff
@@ -221,3 +240,57 @@ func bench(target *prog.Target, file string) {
 	fmt.Printf("program size: min=%v avg=%v max=%v 10%%=%v 50%%=%v 90%%=%v\n",
 		lens[0], sum/n, lens[n-1], lens[n/10], lens[n/2], lens[n*9/10])
 }
+
+func print(file string) {
+	db, err := db.Open(file, false)
+	if err != nil {
+		tool.Failf("failed to open database: %v", err)
+	}
+	keys := maps.Keys(db.Records)
+	sort.Strings(keys)
+	for _, key := range keys {
+		rec := db.Records[key]
+		fmt.Printf("%v\n%v\n", key, string(rec.Val))
+	}
+}
+
+func rm(file, syscall string, target *prog.Target) error {
+	db, err := db.Open(file, false)
+	if err != nil {
+		return fmt.Errorf("failed to open database: %w", err)
+	}
+	for key, rec := range db.Records {
+		result := removeLinesContaining(rec.Val, syscall)
+		if len(result) > 0 {
+			p, err := target.Deserialize(result, prog.NonStrict)
+			if err != nil {
+				return fmt.Errorf("failed to deserialize: %w\n%s", err, result)
+			}
+			if err := p.Validate(); err != nil {
```

Comment:
It's unclear what program this validates. The program we test in the unit test is not the program that passes validation. The program in the test absolutely must not pass validation . The program with the removed first call has dangling resources.
It's possible that Desrialize changed the program to e.g. just "getpid()", and that passed validation, and the test checked for completely different program.
To be able assess what the manager will get after Deserialize, we would need to call p.Serialize on the result and store _that_ to the DB. Then the test would observe what manager will get as the result.

I would suggest to change call removal to this:
```
		p, err := target.Deserialize(data, prog.NonStrict)
		if err != nil { ... }
		for i := range p.Calls - 1; i >= 0; i-- {
			if strings.Contains(p.Calls[i].Meta.Name, syscall) {
				p.RemoveCall(i)
			}
		}
		data = p.Serialize()
		... store it back to db ...
```

Then the test will actually observe and test the resulting program.